component/health.go

```diff
@@ -0,0 +1,31 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//       http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package component // import "go.opentelemetry.io/collector/component"
+
+import (
+	"go.opentelemetry.io/collector/config"
+)
+
+type HealthEvent struct {
+	ComponentID config.ComponentID
+	Error       error
+}
+
+type HealthNotifications interface {
+	Subscribe() <-chan (HealthEvent)
+	Unsubscribe(subscription <-chan (HealthEvent))
+	Send(event HealthEvent)
+	Stop()
+}
```

Comment:
Some questions:
1. Why is this an interface instead of a struct? Should these be on the host directly?
2. I know we discussed about this, but I am not convinced that we need a push systems for this. Majority of the cases are just "tell me the health for foo". The argument I heard that the LB exporter may use this, I think we said that for that case ConsumerError is enough to know (maybe I am wrong, but want to make sure we don't overextend and complicate the API)